glue/sample/src/sinter/_plotting.py

```diff
@@ -337,12 +350,12 @@ def _rescale(v: np.ndarray, scale: str, invert: bool) -> np.ndarray:
 def plot_custom(
         *,
         ax: 'plt.Axes',
-        stats: 'Iterable[sinter.TaskStats]',
-        x_func: Callable[['sinter.TaskStats'], Any],
-        y_func: Callable[['sinter.TaskStats'], Union['sinter.Fit', float, int]],
-        group_func: Callable[['sinter.TaskStats'], TCurveId] = lambda _: None,
-        filter_func: Callable[['sinter.TaskStats'], Any] = lambda _: True,
-        plot_args_func: Callable[[int, TCurveId, List['sinter.TaskStats']], Dict[str, Any]] = lambda index, group_key, group_stats: dict(),
+        stats: Iterable[TaskStats],
+        x_func: Callable[[TaskStats], Any],
+        y_func: Callable[[TaskStats], Union[Fit, float, int]],
+        group_func: _GroupFunc[TCurveId] = lambda _: None,
+        filter_func: Callable[[TaskStats], Any] = lambda _: True,
+        plot_args_func: Union[_PlotArgsFunc2[TCurveId], _PlotArgsFunc3[TCurveId]] = lambda index, group_key, group_stats: dict(),
```

Comment:
This is an example where I sorta prefer for the exposed API to not mention the deprecated functionality, even though the method still technically supports it. I can't remove the backwards compat, but I can hide the complication of the backwards compat from new users.